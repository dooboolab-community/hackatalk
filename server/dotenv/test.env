JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
