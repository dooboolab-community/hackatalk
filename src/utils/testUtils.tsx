import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
