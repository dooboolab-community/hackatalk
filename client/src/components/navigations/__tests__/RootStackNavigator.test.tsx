import 'react-native';

import {createTestElement, createTestProps} from '../../../../test/testUtils';

import React from 'react';
import StackNavigator from '../RootStackNavigator';
import {render} from '@testing-library/react-native';

jest.mock('../../../components/pages/SignIn/SocialSignInButton', () => 'test');

const component = createTestElement(<StackNavigator {...createTestProps()} />);

describe('[Stack] navigator', () => {
  it('should renders without crashing', async () => {
    const json = render(component);

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});
