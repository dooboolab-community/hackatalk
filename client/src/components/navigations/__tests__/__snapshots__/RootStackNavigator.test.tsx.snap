// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Stack] navigator should renders without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#000",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "(0, 0, 0, 0)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        collapsable={false}
        source={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        style={
          Object {
            "borderRadius": 28,
            "opacity": 0.75,
            "transform": Array [
              Object {
                "scale": 1.7,
              },
            ],
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
